java/flight/flight-core/src/main/java/org/apache/arrow/flight/grpc/ServerInterceptorAdapter.java

```diff
@@ -82,7 +82,14 @@ public ServerInterceptorAdapter(List<KeyFactory<?>> factories) {
   @Override
   public <ReqT, RespT> Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, Metadata headers,
       ServerCallHandler<ReqT, RespT> next) {
-    final CallInfo info = new CallInfo(FlightMethod.fromProtocol(call.getMethodDescriptor().getFullMethodName()));
+    FlightMethod method;
+    try {
+      method = FlightMethod.fromProtocol(call.getMethodDescriptor().getFullMethodName());
```

Comment:
I wonder if we could do something like check the service name instead, to avoid catching an exception + to avoid silently passing things through if we add new Flight methods in the future?

Reply:
Hmmm makes sense! I'll see what I can do